tests/unit/query-array-test.js

```diff
@@ -17,7 +17,7 @@ module('unit/query-array', function(hooks) {
     }();
 
     this.createRecordArray = function(options = {}) {
-      return new M3QueryArray(
+      return M3QueryArray.create(
```

Comment:
Wouldn't it be better to drop `ArrayProxy` as the base class?

Reply:
yes please (if possible)

Reply:
Yes, but that might take more work. Will investigate when I get a chance :)